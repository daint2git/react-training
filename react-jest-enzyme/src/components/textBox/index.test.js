import React from 'react'
import { shallow, mount, render } from 'enzyme'
import renderer from 'react-test-renderer'
import TextBox from './'

describe('TextBox', () => {
  test('matches the snapshot', () => {
    const wrapper = shallow(<TextBox />)
    expect(wrapper).toMatchSnapshot()
  })

  test('onChange handle', () => {
    const props = {
      value: '',
      onChange: jest.fn(),
    }
    const wrapper = shallow(<TextBox {...props} />)
    const input = wrapper.find('input[type="text"]')
    expect(input.props().value).toBe('')
    // input.simulate('change', { target: { value: 'Hello' } })
    // expect(props.onChange).toHaveBeenCalled()
    // expect(props.onChange).toHaveBeenCalledWith({ target: { value: 'Hello' } })
    // wrapper.update()
    // const input2 = wrapper.find('input[type="text"]')
    // console.log(input2.props())
    // expect(wrapper.find('input[type="text"]').text()).toBe('hello')
  })
})
