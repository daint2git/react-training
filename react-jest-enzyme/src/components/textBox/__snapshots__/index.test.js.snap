// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox matches the snapshot 1`] = `
<input
  type="text"
/>
`;
